fishtank/src/backend/docker.ts

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { ChildProcess, execFile, ExecFileOptions } from 'child_process'
+import { promisify } from 'util'
+
+const DEFAULT_COMMAND = 'docker'
+
+type ExecFilePromiseReturn = { child: ChildProcess; stdout: string; stderr: string }
+
+type ExecFilePromiseError = {
+  cmd: string
+  code: number | string
+  stdout: string
+  stderr: string
+}
+
+const execChildProcess = promisify<
```

Comment:
`const execFileChildProcess` since this is promisify the `execFile` function

Reply:
I would prefer the name stick with Nodejs's convention since we are not adding logic here, just directly calling that function with promisify. This way, in the future if we need to call exec, it will be easy to add and not cause any confusion.

Reply:
`exec` should never be used in our code because it constitutes a security threat. This is also explicitly called out in the documentation:

> Never pass unsanitized user input to this function. Any input containing shell metacharacters may be used to trigger arbitrary command execution.

and this can be seen by playing with a few examples.

Having said that, ok to rename to `execFile`, even though it's not the original `execFile`.

Reply:
I mean I prefer `execFilePromisified` which has the prefix of Nodejs function. In the future we can add `execPromisified`.